### PR TITLE
arm gicv2 hyper ext

### PR DESCRIPTION
**Pull request by pengzechen**

### CODE REVIEW

File: src/gic_v2.rs [hunk 1]
```diff
@@ -9,8 +9,11 @@ use tock_registers::interfaces::{Readable, Writeable};
 use tock_registers::register_structs;
 use tock_registers::registers::{ReadOnly, ReadWrite, WriteOnly};
 
+use crate::GIC_CONFIG_BITS;
+use crate::SGI_RANGE;
+
 register_structs! {
-    /// GIC Distributor registers.
+
```
Comment (hky1999) [inline src/gic_v2.rs]: Why did you delete this line ：）

File: src/gic_v2.rs [hunk 2]
```diff
@@ -188,7 +335,7 @@ impl GicDistributor {
         assert!(max_irqs <= GIC_MAX_IRQ);
         self.max_irqs = max_irqs;
 
-        // Disable all interrupts
+        /// Disable all interrupts
```
Comment (hky1999) [inline src/gic_v2.rs]: use `//`
ref: https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text

File: src/gic_v2.rs [hunk 3]
```diff
@@ -199,12 +346,12 @@ impl GicDistributor {
                 self.regs().ITARGETSR[i / 4].set(0x01_01_01_01);
             }
         }
-        // Initialize all the SPIs to edge triggered
+        /// Initialize all the SPIs to edge triggered
```
Comment (hky1999) [inline src/gic_v2.rs]: use `//`

File: src/gic_v2.rs [hunk 4]
```diff
@@ -199,12 +346,12 @@ impl GicDistributor {
...
         for i in SPI_RANGE.start..max_irqs {
             self.configure_interrupt(i, TriggerMode::Edge);
         }
 
-        // enable GIC0
+        /// enable GIC0
```
Comment (hky1999) [inline src/gic_v2.rs]: use `//`

File: src/lib.rs [hunk 5]
```diff
@@ -29,6 +29,8 @@ pub const SPI_RANGE: Range<usize> = 32..1020;
 /// Maximum number of interrupts supported by the GIC.
 pub const GIC_MAX_IRQ: usize = 1024;
 
+pub const GIC_CONFIG_BITS: usize = 2;
```
Comment (hky1999) [inline src/lib.rs:33]: add comments here

**Review by hky1999 — CHANGES_REQUESTED** [review]

File: src/gic_v2.rs (hunk 4 above)
Comment (pengzechen) [reply src/gic_v2.rs]: ok

File: src/gic_v2.rs (hunk 3 above)
Comment (pengzechen) [reply src/gic_v2.rs]: ok

File: src/gic_v2.rs (hunk 2 above)
Comment (pengzechen) [reply src/gic_v2.rs]: done

File: src/gic_v2.rs (hunk 1 above)
Comment (pengzechen) [reply src/gic_v2.rs]: It should be added.

File: src/gic_v2.rs [hunk 6]
```diff
@@ -267,8 +431,14 @@ impl GicCpuInterface {
     ///
     /// This function should be called only once.
     pub fn init(&self) {
-        // enable GIC0
```
Comment (hky1999) [inline src/gic_v2.rs:270]: recover this comment

**Review by hky1999 — APPROVED** [review]

**Comment by hky1999** [comment]
Remember to squash your commits.

File: src/gic_v2.rs (hunk 6 above)
Comment (pengzechen) [reply src/gic_v2.rs:270]: ok
